lib/fauna/page.rb

```diff
@@ -221,6 +221,38 @@ def reverse_each
       end
     end
 
+    ##
+    # Returns the flattened contents of the set as an array.
+    #
+    # Ideal for when you need the full contents of a set with a known small size. If you need to iterate over a set
+    # of large or unknown size, it is recommended to use +each+ instead.
+    #
+    # The set is paged in the +after+ direction.
+    def set_contents
+      each.flat_map { |x| x }
+    end
+
+    ##
+    # Iterates over the entire set, applying the configured map and discarding the result.
+    #
+    # Ideal for performing a +foreach+ over an entire set (like deleting all instances in a set). The set is iterated in
+    # the +after+ direction.
+    #
+    # Example of deleting every instance in a set:
+    #
+    #   page.apply_map! { |page_q| foreach(page_q) { |ref| delete ref } }
+    def apply_map!(&block)
```

Comment:
There are `map` `foreach` and `filter`. I think the best thing would be to make sure we mirror the relevant parts of the API. 

I'm not sure how I feel about the proliferation of `with` in all of these.... it adds a lot of noise esp if many of these are chained. Perhaps we should just drop `with_`.